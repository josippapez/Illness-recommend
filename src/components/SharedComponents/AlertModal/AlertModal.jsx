import React from 'react';
import PropTypes from 'prop-types';

import './AlertModal.scss';
import Modal from '../Modal/Modal';

const AlertModal = props => {
  return (
    <Modal className='modal-for-alert'>
      <div className='modal-container-alert'>
        <div className='alert-modal'>
          <div className='alert-modal__icon-alert'></div>
          <div className='alert-modal__header-text'>
            <div className='header-main-text'>Oprez!</div>
            <div className='header-small-text'>{props.alertInfotext}</div>
          </div>
          <div className='button-container'>
            {!props.showOnlyConfirmOption && (
              <button
                className='alert-decline-button'
                onClick={props.declineOptions}
              >
                Ne
              </button>
            )}
            <button
              className='alert-confirm-button'
              onClick={props.confirmOptions}
            >
              {props.showOnlyConfirmOption ? 'U redu' : 'Da'}
            </button>
          </div>
        </div>
      </div>
    </Modal>
  );
};

AlertModal.propTypes = {
  alertInfotext: PropTypes.string,
  confirmOptions: PropTypes.func,
  declineOptions: PropTypes.func,
  showOnlyConfirmOption: PropTypes.bool,
};

export default AlertModal;
