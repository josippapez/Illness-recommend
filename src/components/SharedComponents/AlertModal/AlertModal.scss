.modal-for-alert {
  .overlay {
    z-index: 2;
  }
  .modal-container-alert {
    z-index: 5;
    width: 320px;
    background: var(--white) 0% 0% no-repeat padding-box;
    box-shadow: 0px 10px 20px #00000034;
    border-radius: 6px;
    padding: 20px 12px 12px 12px;

    .alert-modal {
      display: flex;
      flex-direction: column;
      align-items: center;
      text-align: center;
      &__icon-alert {
        background-image: url("../../../styles/assets/images/AlertIconwFrame.svg");
        background-repeat: no-repeat;
        width: 38px;
        height: 38px;
      }
      &__header-text {
        margin-top: 10px;
        margin-bottom: 30px;
        .header-main-text {
        }
        .header-small-text {
          margin-top: 10px;
          font: normal normal normal 13px/19px "Lato, Regular";
          letter-spacing: 0px;
          color: #474747;
        }
      }
      .button-container {
        display: flex;
        flex-direction: row;
        .alert-decline-button {
          width: 146px;
          height: 38px;
          background: #00000020 0% 0% no-repeat padding-box;
          border-radius: 4px;
          border: none;
          color: var(--main-text);
          font: normal normal bold 14px/17px "Lato, Regular";
          letter-spacing: 0;
          margin-right: 10px;
          &:hover {
            cursor: pointer;
            background: #0000004b 0% 0% no-repeat padding-box;
          }
        }
        .alert-confirm-button {
          color: var(--white);
          letter-spacing: 0px;
          border: none;
          width: 144px;
          height: 38px;
          background: var(--hyperlink) 0% 0% no-repeat padding-box;
          border-radius: 4px;
          font: normal normal bold 14px/17px "Lato, Regular";
          &:hover {
            cursor: pointer;
            background: #0063c0 0% 0% no-repeat padding-box;
          }
        }
      }
    }
  }
}
