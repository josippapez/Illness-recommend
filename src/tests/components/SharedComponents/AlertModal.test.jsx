/* eslint-disable no-undef */
import React from 'react';
import { mount } from 'enzyme';
import AlertModal from '../../../components/SharedComponents/AlertModal/AlertModal';

const props = {
  alertInfotext: 'test',
  confirmOptions: jest.fn(),
  declineOptions: jest.fn(),
  showOnlyConfirmOption: false,
};

describe('Alert Modal', () => {
  let component;

  const modalRoot = global.document.createElement('div');
  modalRoot.setAttribute('id', 'root');
  const body = global.document.querySelector('body');
  body.appendChild(modalRoot);

  afterEach(() => {
    component.unmount();
  });

  it('Renders correctly', () => {
    component = mount(<AlertModal {...props} />);
    expect(component.html()).toMatchSnapshot();
  });
});
