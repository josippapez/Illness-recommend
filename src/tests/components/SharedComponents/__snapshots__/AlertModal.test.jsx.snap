// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Modal Renders correctly 1`] = `"<div class=\\"modal modal-for-alert\\"><div class=\\"overlay\\"><div class=\\"overlay__children\\"><div class=\\"modal-container-alert\\"><div class=\\"alert-modal\\"><div class=\\"alert-modal__icon-alert\\"></div><div class=\\"alert-modal__header-text\\"><div class=\\"header-main-text\\">Oprez!</div><div class=\\"header-small-text\\">test</div></div><div class=\\"button-container\\"><button class=\\"alert-decline-button\\">Ne</button><button class=\\"alert-confirm-button\\">Da</button></div></div></div></div></div></div>"`;
